test/integration/analysis.js

```diff
@@ -120,6 +123,52 @@ describe('workflow', function() {
             });
         });
 
+        it('PRECHECK query should be inside a read-only transaction', function(done) {
+            var TEST_SOURCE_TYPE = 'test-source';
+            var TestSource = Node.create(TEST_SOURCE_TYPE, {
+                table: Node.PARAM.STRING()
+            }, {cache: true});
+            TestSource.prototype.sql = function() {
+                return 'select * from ' + this.table;
+            };
+            TestSource.prototype.preCheckQuery = function() {
+                return 'INSERT INTO ' + this.table + ' VALUES (999, NULL, NULL, 100)';
+            };
+
+            var supportsTypeFn = Factory.prototype._supportsType;
+            var getNodeClassFn = Factory.prototype.getNodeClass;
+
+            Factory.prototype._supportsType = function() {
+                return true;
+            };
+
+            Factory.prototype.getNodeClass = function() {
+                return TestSource;
+            };
+
+            var definition = {
+                type: TEST_SOURCE_TYPE,
+                params: {
+                    table: 'airbnb_rooms'
+                }
+            };
+            testConfig.batch.inlineExecution = true;
+            Analysis.create(testConfig, definition, function(err, analysis) {
+                Factory.prototype._supportsType = supportsTypeFn;
+                Factory.prototype.getNodeClass = getNodeClassFn;
+                assert.ifError(err);
+                var catalogQuery = ['SELECT last_error_message ',
```

Comment:
@rochoa I didn't find a way to get the last message which didn't involve to query the analysis catalog table. I'm pretty sure it exists.

The test is a bit complex this way but I think we shouldn't extract it unless we are going to make the same in other tests